src/main/java/build/buildfarm/common/Actions.java

```diff
@@ -36,13 +36,23 @@ public static String invalidActionMessage(Digest actionDigest) {
     return format("Action %s is invalid", DigestUtil.toString(actionDigest));
   }
 
+  public static String invalidActionVerboseMessage(
+      Digest actionDigest, PreconditionFailure failure) {
+    return format(
+        "Action %s is invalid: %s",
+        DigestUtil.toString(actionDigest),
+        failure.getViolationsList().stream()
+            .map(Violation::getDescription)
+            .reduce("", (message, description) -> message + description + " "));
```

Comment:
The list of violations can get *very* long - we need to ensure that this is trimmed to something reasonable: 3 violations with an ellipses is my preference.